lib/widget/calendar.dart

```diff
@@ -0,0 +1,474 @@
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:yaml/yaml.dart';
+
+final kToday = DateTime.now();
+final kFirstDay = DateTime(kToday.year, kToday.month - 3, kToday.day);
+final kLastDay = DateTime(kToday.year, kToday.month + 3, kToday.day);
+
+class EnsembleCalendar extends StatefulWidget
+    with Invokable, HasController<CalendarController, CalendarState> {
+  static const type = 'Calendar';
+  EnsembleCalendar({Key? key}) : super(key: key);
+
+  final CalendarController _controller = CalendarController();
+  @override
+  CalendarController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => CalendarState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'selectedCell': () => _controller.selectedDays.value.toList(),
+      'markedCell': () => _controller.markedDays.value.toList(),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'markCall': (value) => _markCell(value),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'rowHeight': (value) =>
+          _controller.rowHeight = Utils.getDouble(value, fallback: 52.0),
+      'cell': (value) => setCellData(value, _controller.cell),
+      'selectedCell': (value) => setCellData(value, _controller.selectedCell),
+      'todayCell': (value) => setCellData(value, _controller.todayCell),
+      'range': (value) => setRangeData(value),
+    };
+  }
+
+  void _markCell(dynamic value) {
+    final date = Utils.getDate(value);
+    if (date == null) return;
+    _controller.markedDays.value.add(date);
+  }
+
+  void setRangeData(dynamic data) {
+    if (data is YamlMap) {
+      _controller.rangeSelectionMode = RangeSelectionMode.toggledOn;
+      _controller.onRangeStart =
+          EnsembleAction.fromYaml(data['onStart'], initiator: this);
+      _controller.onRangeComplete =
+          EnsembleAction.fromYaml(data['onComplete'], initiator: this);
+      setCellData(data['startCell'], _controller.rangeStartCell);
+      setCellData(data['endCell'], _controller.rangeEndCell);
+    }
+  }
+
+  void setCellData(dynamic data, Cell cell) {
+    if (data is YamlMap) {
+      cell.widget = data['widget'];
+      cell.onTap = EnsembleAction.fromYaml(data['onTap'], initiator: this);
+      if (data.containsKey('config')) {
+        cell.config = CellConfig(
+            color: Utils.getColor(data['config']['color']),
+            padding: Utils.getInsets(data['config']['padding']),
+            margin: Utils.getInsets(data['config']['margin']),
+            labelStyle: Utils.getTextStyle(data['config']['labelStyle']),
+            alignment: Utils.getAlignment(data['config']['alignment']),
+            borderRadius:
+                Utils.getBorderRadius(data['config']['borderRadius']));
+      }
+    }
+  }
+}
+
+class CellConfig {
+  Color? color;
```

Comment:
need textStyle

Reply:
their is labelStyle, I'll update it to textStyle.